GVFS/GVFS.Virtualization/FileSystem/FileSystemVirtualizer.cs

```diff
@@ -269,18 +269,17 @@ protected void OnHardLinkCreated(string relativeExistingFilePath, string relativ
         {
             try
             {
-                if (!string.IsNullOrEmpty(relativeNewLinkPath))
+                bool newLinkPathInDotGit = FileSystemCallbacks.IsPathInsideDotGit(relativeNewLinkPath);
+                bool relativePathInDotGit = FileSystemCallbacks.IsPathInsideDotGit(relativeExistingFilePath);

```

Comment:
Rename `relativePathInDotGit` to something like `existingFilePathInDotGit` to make it clearer which path we're referring to?